src/main/java/sirius/kernel/commons/StringCleanup.java

```diff
@@ -316,6 +461,112 @@ public static String uppercase(@Nonnull String input) {
+        return STRIP_XML_REGEX.matcher(input).replaceAll("");
+    }
+
+    /**
+     * Removes all unsafe HTML tags from the given string.
+     *
+     * @param input the input to process
+     * @return the resulting string
+     */
+    @Nonnull
+    public static String removeUnsafeHtml(@Nonnull String input) {
+        return STRIP_XML_REGEX.matcher(input)
+                              .replaceAll(match -> Strings.DETECT_ALLOWED_HTML_REGEX.matcher(match.group()).matches() ?
+                                                   match.group() :
+                                                   "");
+    }
+
+    /**
+     * Resolves encoded HTML entities to their plain text equivalent in the given string.
+     *
+     * @param input the input to process
+     * @return the resulting string
+     */
+    @Nonnull
+    public static String decodeHtmlEntities(@Nonnull String input) {
+        input = input.replace("&nbsp;", " ")
+                     .replace("&auml;", "ä")
+                     .replace("&ouml;", "ö")
+                     .replace("&uuml;", "ü")
+                     .replace("&Auml;", "Ä")
+                     .replace("&Ouml;", "Ö")
+                     .replace("&Uuml;", "Ü")
+                     .replace("&szlig;", "ß")
+                     .replace("&lt;", "<")
+                     .replace("&gt;", ">")
+                     .replace("&quot;", "\"")
+                     .replace("&apos;", "'")
+                     .replace("&amp;", "&")
+                     .replace("&#8226; ", "* ")
+                     .replace("&#8226;", "* ")
+                     .replace("&#8227; ", "* ")
+                     .replace("&#8227;", "* ")
+                     .replace("&#8259; ", "* ")
+                     .replace("&#8259;", "* ");
+        input = Strings.replaceAll(PATTERN_DECIMAL_ENTITY, input, s -> {
+            try {
+                return String.valueOf(Character.toChars(Integer.parseInt(s)));
+            } catch (NumberFormatException e) {
+                Exceptions.ignore(e);
+            } catch (Exception e) {
+                Exceptions.handle(e);
+            }
+            return "";
+        });
+        input = Strings.replaceAll(PATTERN_HEX_ENTITY, input, s -> {
+            try {
+                return String.valueOf(Character.toChars(Integer.parseInt(s, 16)));
+            } catch (NumberFormatException e) {
+                Exceptions.ignore(e);
+            } catch (Exception e) {
+                Exceptions.handle(e);
+            }
+            return "";
+        });
+
+        return input;
+    }
+
+    /**
+     * Normalizes a text by removing all HTML, entities and special characters.
+     *
+     * @param input the input to process
+     * @return the resulting string
+     */
+    @Nonnull
+    public static String htmlToPlainText(@Nonnull String input) {
+        String normalizedText = input;
+
+        if (STRIP_XML_REGEX.matcher(normalizedText).find()) {
```

Comment:
if (!Strings.containsAllowedHtml(input)) {
return input;
}

Reply:
If anything it would rather use `Strings.probablyContainsXml` as this method does not have anything to do with the allowed tags.